ilastik/experimental/parser/_base.py

```diff
@@ -1,137 +1,41 @@
-import json
-import pickle
-from typing import Any, List, Optional
+from typing import Optional
 
 import h5py
-import numpy
 
 from . import types
 
-PIXEL_CLASSIFICATION = b"Pixel Classification"
 
+class PixelClassificationProject:
+    ILP_WORKFLOW_NAME = b"Pixel Classification"
 
-class IlastikProject:
-    def __init__(self, path: str, mode: str = "r") -> None:
-        if mode != "r":
-            raise NotImplementedError(f"mode: {mode}")
-
-        self.__mode = mode
-        self.__path = path
-        self.__file = None
-
-    def __enter__(self):
-        self.__file = h5py.File(self.__path, mode=self.__mode)
-        return _create_project_wrap(self.__file)
-
-    def __exit__(self, exc_type, exc, tb):
-        if self.__file:
-            self.__file.close()
-
-
-WORKFLOW_KEY = "workflowName"
-
-
-def _create_project_wrap(hdf5_file):
-    type_ = hdf5_file[WORKFLOW_KEY][()]
-    if type_ == PIXEL_CLASSIFICATION:
-        return _PixelClassProjectImpl(hdf5_file)
-
-    raise NotImplementedError(f"Unknown project type {type_}")
-
-
-class _Keys:
-    INPUT_DATA = "Input Data"
-    INPUT_DATA_INFOS = "infos"
-    INPUT_DATA_RAW = "Raw Data"
-    INPUT_DATA_AXIS_TAGS = "axistags"
-    INPUT_DATA_SHAPE = "shape"
-
-    FEATURES = "FeatureSelections"
-    FEATURES_IDS = "FeatureIds"
-    FEATURES_SCALES = "Scales"
-    FEATURES_COMPUTE_IN_2D = "ComputeIn2d"
-    FEATURES_SELECTION_MATRIX = "SelectionMatrix"
-    PIXEL_CLASSIFICATION = "PixelClassification"
-    PIXEL_CLASSIFICATION_TYPE = "pickled_type"
-    PIXEL_CLASSIFICATION_FACTORY = "ClassifierFactory"
-    PIXEL_CLASSIFICATION_FORESTS = "ClassifierForests"
-    LABEL_NAMES = "LabelNames"
+    def __init__(self, hdf5_file: h5py.File) -> None:
```

Comment:
If this is supposed to represent data from a project file, I'd vote for converting this into a dataclass with constructor taking `data_info`, `feature_matrix`, `classifier` etc. and adding classmethod or external function for parsing it from HDF5 file.